### ISSUE TITLE
errors when running with keras as a backend

### ISSUE THREAD

**Issue by 92wyunchao** [ChrisCummins/phd#52]
Hi,
When I run the commad with the keras as a backend: 
bazel run //deeplearning/clgen --     --config $PWD/deeplearning/clgen/tests/data/tiny/config.pbtxt

I get the following errors:
INFO: Analysed target //deeplearning/clgen:clgen (0 packages loaded).
INFO: Found 1 target...
Target //deeplearning/clgen:clgen up-to-date:
  bazel-phd/bazel-out/k8-py3-opt/bin/deeplearning/clgen/clgen
INFO: Elapsed time: 0.315s, Critical Path: 0.01s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
I0927 20:20:18.668901 140597635225408 corpuses.py:163] Content ID: 6b4248154caa140665461619e4d1292eeee1b225
I0927 20:20:18.680049 140597635225408 preprocessed.py:154] Content files: 803,729 chars, 43,143 lines, 250 files.
I0927 20:20:18.680175 140597635225408 preprocessed.py:157] Pre-processed 250 files in 43,442 ms (1.99x speedup).
I0927 20:20:18.680385 140597635225408 preprocessed.py:160] Pre-processing discard rate: 32.8% (82 files).
I0927 20:20:18.680593 140597635225408 preprocessed.py:163] Pre-processed corpus: 759,966 chars, 42,480 lines, 168 files.
I0927 20:20:18.684421 140597635225408 corpuses.py:180] AsciiCharacterAtomizer: 92 tokens in 0 ms
I0927 20:20:18.689182 140597635225408 encoded.py:153] Encoded 168 files in 28 ms (2.29x speedup).
I0927 20:20:18.689357 140597635225408 encoded.py:155] Encoded corpus: 759,966 tokens, 168 files.
/root/.cache/bazel/_bazel_root/ce788644d82188e468bbfba809395ca6/execroot/phd/bazel-out/k8-py3-opt/bin/deeplearning/clgen/clgen.runfiles/pypi__h5py_2_7_1/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.
I0927 20:20:20.135394 140597635225408 keras_backend.py:75] Model summary:
_________________________________________________________________
Layer (type)                 Output Shape              Param #
=================================================================
embedding_1 (Embedding)      (64, 64, 64)              5888
_________________________________________________________________
dropout_1 (Dropout)          (64, 64, 64)              0
_________________________________________________________________
lstm_1 (LSTM)                (64, 64, 128)             98816
_________________________________________________________________
dropout_2 (Dropout)          (64, 64, 128)             0
_________________________________________________________________
lstm_2 (LSTM)                (64, 64, 128)             131584
_________________________________________________________________
dropout_3 (Dropout)          (64, 64, 128)             0
_________________________________________________________________
time_distributed_1 (TimeDist (64, 64, 92)              11868
=================================================================
Total params: 248,156
Trainable params: 248,156
Non-trainable params: 0
_________________________________________________________________

E0927 20:20:20.136460 140597635225408 **clgen.py:176] 'Corpus' object has no attribute 'decoder' (AttributeError)**

  stacktrace:
      #1  /root/.cache/bazel/_bazel_root/ce788644d82188e468bbfba809395ca6/execroot/phd/bazel-out/k8-py3-opt/bin/deeplearning/clgen/clgen.runfiles/phd/deeplearning/clgen/models/keras_backend.py:92 Train()
      #2  /root/.cache/bazel/_bazel_root/ce788644d82188e468bbfba809395ca6/execroot/phd/bazel-out/k8-py3-opt/bin/deeplearning/clgen/clgen.runfiles/phd/deeplearning/clgen/models/models.py:154 Train()
      #3  /root/.cache/bazel/_bazel_root/ce788644d82188e468bbfba809395ca6/execroot/phd/bazel-out/k8-py3-opt/bin/deeplearning/clgen/clgen.runfiles/phd/deeplearning/clgen/models/models.py:194 Sample()
      #4  /root/.cache/bazel/_bazel_root/ce788644d82188e468bbfba809395ca6/execroot/phd/bazel-out/k8-py3-opt/bin/deeplearning/clgen/clgen.runfiles/phd/deeplearning/clgen/clgen.py:281 DoFlagsAction()
      #5  /root/.cache/bazel/_bazel_root/ce788644d82188e468bbfba809395ca6/execroot/phd/bazel-out/k8-py3-opt/bin/deeplearning/clgen/clgen.runfiles/phd/deeplearning/clgen/clgen.py:205 RunContext()

**Comment by ChrisCummins** [comment]
Hi @92wyunchao, unfortunately, the Keras background is totally abandoned and never really got to a usable state. I'll remove it in the next release. Use the Tensorflow backend instead :)

Cheers,
Chris
